run.py

```diff
@@ -17,11 +21,25 @@ def install_and_import(package):
         globals()[package] = importlib.import_module(package)
 
 
+# https://stackoverflow.com/questions/3764291/checking-network-connection
+def have_internet():
+    conn = httplib.HTTPConnection("www.google.com", timeout=5)
+    try:
+        conn.request("HEAD", "/")
+        conn.close()
+        return True
+    except:
+        conn.close()
+        return False
```

Comment:
Couldn't both these `conn.close()` be removed and a single call just be in a `finally` clause?